### ISSUE TITLE
AOS does not work when JS is disabled

### ISSUE THREAD

**Issue by marekka** [michalsnik/aos#722]
Hi, AOS does not work when JS is disabled, think it's a major issue!

Are you going to fix this?

**Comment by MikeyJN** [comment]
Considering the plug-in relies on JS to detect when elements are moving in and out of the viewport, I think you'd have a tough time to 'fix' it without it lol

Throw in the fact that 95%+ of websites use JS in one way or another I'd struggle to think of a reason why anyone would ever want to disable JS in the first place?!

**Comment by websterchak** [comment]
I don't think it will work without JS (Javascript). In fact, this is a JS Library which, as it named, would require JS enabled (very common) to work. Without it, it'd be like expecting a car to move itself without engine!

**Comment by marekka** [comment]
Well, IMO, there should be code snippets to satisfy no-js browsers and have
the AOS section always visible onload

Z wyrazami szacunku
Marek Kaczanowicz

[image: Mailtrack]
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
Sender
notified by
Mailtrack
<https://mailtrack.io?utm_source=gmail&utm_medium=signature&utm_campaign=signaturevirality11&>
04.11.21,
21:24:02

wt., 2 lis 2021 o 04:29 Webster Chak ***@***.***> napisał(a):

**Comment by scriptex** [comment]
You can always implement this yourself. I agree with what @MikeyJN said and it really should be out of the scope of any JS library to provide support for users with disabled JS.

What you can do is:
1. Conditionally load the CSS for AOS with JS.
2. Add a classname to your `<body>` or `<html>` tag and then tell the CSS to show all `[data-aos]` elements when this classname is present. then remove the classname with JS.
3. Anything else which you find suitable for your use case, library, framework, etc. The possibilities are endless.

**Comment by Vuurvos1** [comment]
While going through the project recently I found [this](https://github.com/michalsnik/aos/blob/next/src/sass/_animations.scss#L5) line of scss which checks if a `.no-js` class is present on the html element, so the author did think of users without js enabled.
But to take advantage of this you would need to remove this class inside your own javascript.

A better solution could be for the library to add a custom class to the html once it initializes which the styles then could reference.

**Comment by marekka** [comment]
Have to be something legacy.... I have analysed CSS files connected to the demo:

https://michalsnik.github.io/aos/

And they do not have any references to no-js......

**Comment by mr-cheffy** [comment]
they can't fix it since the library relies on js. What I suggest is to use a `<noscript>` tag to tell the user to allow JS...

**Comment by domdomegg** [comment]
If you want things to display as default if the user doesn't have JavaScript enabled, you can use the following snippets:

You can add this to your HTML:

```html
<noscript>
  <style>[data-aos] { opacity: 1 !important; transform: none !important; }</style>
</noscript>
```

Or, in JSX/TSX (React):

```jsx
<noscript>
  <style>{"[data-aos] { opacity: 1 !important; transform: none !important; }"}</style>
</noscript>
```

**Comment by marekka** [comment]
THANKS!

**Comment by seppebeelprez** [comment]
I tried the <noscript> way but Google still says there's something that eliminates render-blocking resources. 

What did work is adding a class for exampla "no-js" to the html tag and removing it with a inline script right after the body tag. Removing the no-js class with jQuery won't work as I'm still getting that render-blocking error.

- So default I added "no-js" to the html tag
- Right after the body tag you can include this: `<script>
  document.querySelector('html').classList.remove('no-js');
</script>`
- And this is what I added as css to the html tag: `html {
  &.no-js {
    [data-aos] {
      opacity: 1 !important;
      transform: none !important;
    }
  }
}`

I'm not sure why the <noscript> tag doesn't work with Google because it definitely works if I disable javascript in Firefox.
